specs/nimble-blazor/blazor-net8-upgrade-hld.md

```diff
@@ -0,0 +1,88 @@
+# .NET 8 Support for Nimble/Spright Blazor
+
+## Problem Statement
+
+Currently our Blazor projects (Nimble/Spright) target .NET 6, which will reach [end-of-support on November 12, 2024](https://devblogs.microsoft.com/dotnet/dotnet-6-end-of-support/).
+
+Before that date, we should update our Blazor projects to target .NET 8, which will remain in-support until November 2026 ([more info on versions / support duration here](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core)).
+
+
+## Links To Relevant Work Items and Reference Material
+
+Design System work items:
+- [Feature 2812085: Nimble Blazor .NET 8 upgrade](https://dev.azure.com/ni/DevCentral/_workitems/edit/2812085)
+- [nimble#1667: Blazor .NET 8 support](https://github.com/ni/nimble/issues/1667)
+
+Related work items:
+- ASW: [Epic 2753078: ASW .NET 8 Migration](https://dev.azure.com/ni/DevCentral/_workitems/edit/2753078)
+
+## Implementation / Design
+
+### Planned Changes ###
+- For all projects in the Nimble `BlazorWorkspace.sln` (including Nimble Blazor and Spright Blazor):
+  - Update `TargetFramework` from `net6.0` to `net8.0`
+  - Update `Microsoft.AspNetCore.Components.Web` references from `6.x` to `8.x`
+  - Update code analyzer package versions to latest versions
+  - Resolve build warnings (mostly due to new code analysis rules)
+  - Update/recreate demo projects based on newer Blazor .NET 8 templates
+  - Ensure adequate test coverage, especially of interactions with Blazor initialization / lifecycle code.
```

Comment:
My understanding is that a big change with .NET 8 is being able to mix and match [render modes](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/render-modes?view=aspnetcore-8.0#render-modes) of SSR, Blazor Server, and Blazor WASM. And also that Razor Class Libraries in [ASP.NET Core MVC and Razor Pages can mix and match with Blazor](https://learn.microsoft.com/en-us/aspnet/core/tutorials/choose-web-ui?view=aspnetcore-8.0#combine-multiple-web-ui-solutions-aspnet-core-mvc-or-razor-pages-plus-blazor). And then there is also Blazor Hybrid.

Since NimbleBlazor really is a Razor Class Library fundamentally, I think it would be helpful to spell out more concretely the different modes we will support and test. My first reaction is that I'd like to support all of those modes.

Reply:
Good point. Called out the Blazor render modes more specifically in the HLD. The example updates will be part of that, but we'll also document which modes we've tested / that we support.

Reply:
As far as mixing&matching Blazor with other MVC/Razor pages, I'm not sure that there's really any additional testing to do for that, since the Blazor example projects are already ASP.NET Core MVC too.